rapids-cmake/test/generate_resource_spec.cmake

```diff
@@ -46,7 +46,19 @@ files to be generated.
 function(rapids_test_generate_resource_spec DESTINATION filepath)
   list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.test.generate_resource_spec")
 
-  if(NOT DEFINED CMAKE_CUDA_COMPILER AND NOT DEFINED CMAKE_CXX_COMPILER)
+  unset(rapids_lang)
+  get_property(rapids_languages GLOBAL PROPERTY ENABLED_LANGUAGES)
+  if("CXX" IN_LIST rapids_languages)
+    set(rapids_lang CXX)
+    # Even when the CUDA language is disabled we want to pass this since it is used by
+    # find_package(CUDAToolkit) to find the location
+    set(CMAKE_TRY_COMPILE_PLATFORM_VARIABLES CMAKE_CUDA_COMPILER)
```

Comment:
I just checked the CMake source code, expecting that `CMAKE_CUDA_COMPILER` would already be propagated by `try_compile()` - it is not. That blows my mind. Surely there must be a reason?

Reply:
IIRC the `try_compile` leverages the `CMakeFiles/` directory of the parent process. So therefore only enabled languages would be propagated.